tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,563 @@
+    metric_idx = 1 + (n_outputs_ if n_outputs_ > 1 else 0)
+
+    # List of lists of metrics
+    if n_outputs_ == 1:
+        metrics_ = [metric_class(name="1"), metric_class(name="2")]
+    else:
+        metrics_ = [
+            [metric_class(name="1"), metric_class(name="2")]
+            for _ in range(n_outputs_)
+        ]
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=metrics_,
+    )
+    est.fit(X, y)
+    if n_outputs_ == 1:
+        assert est.model_.metrics[metric_idx].name == "1"
+    else:
+        # For multi-output models, Keras pre-appends the output name
+        assert est.model_.metrics[metric_idx].name == "out1_1"
+
+    # List of lists of metrics
+    if n_outputs_ == 1:
+        metrics_ = {"out1": [metric_class(name="1"), metric_class(name="2")]}
+    else:
+        metrics_ = {
+            f"out{i+1}": [metric_class(name="1"), metric_class(name="2")]
+            for i in range(n_outputs_)
+        }
+
+    # Dict of metrics
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=metrics_,
+    )
+    est.fit(X, y)
+    if n_outputs_ == 1:
+        assert est.model_.metrics[metric_idx].name == "1"
+    else:
+        # For multi-output models, Keras pre-appends the output name
+        assert est.model_.metrics[metric_idx].name == "out1_1"
+
+
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_metrics_routed_params_iterable(n_outputs_):
+    """Tests compiling metrics with routed parameters
+    when they are passed as an iterable.
+    """
+
+    metrics = metrics_module.BinaryAccuracy
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # loss functions for each output and joined show up as metrics
+    metric_idx = 1 + (n_outputs_ if n_outputs_ > 1 else 0)
+
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=[metrics],
+        metrics__0__name="custom_name",
+    )
+    est.fit(X, y)
+    if n_outputs_ == 1:
+        assert est.model_.metrics[metric_idx].name == "custom_name"
+    else:
+        assert est.model_.metrics[metric_idx].name == "out1_custom_name"
+
+    if n_outputs_ == 1:
+        metrics_ = [
+            metrics,
+        ]
+    else:
+        metrics_ = [metrics for _ in range(n_outputs_)]
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=metrics_,
+        metrics__name="name_all_metrics",  # ends up in index 1 only
+        metrics__0__name="custom_name",  # ends up in index 0 only
+    )
+    est.fit(X, y)
+    if n_outputs_ == 1:
+        assert est.model_.metrics[metric_idx].name == "custom_name"
```

Comment:
Suggested change:

``` diff
-          assert est.model_.metrics[metric_idx].name == "custom_name"
+         metrics = est.model_metrics
+         assert model.metrics[metric_idx].name == "custom_name"
```

This would make the test a little easier to parse.

Reply:
yep